framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -202,8 +206,27 @@ public Void visitTypeCast(TypeCastTree node, Void p) {
             if (castType.getKind() == TypeKind.LONG && castRange.isLongEverything()) {
                 return p;
             }
+            if (Range.IGNORE_OVERFLOW) {
+                // Range.IGNORE_OVERFLOW is only set if this checker is ignoring overflow.
+                // In that case, compress casts to the range of the underlying type.
```

Comment:
This is necessary to suppress `cast.unsafe` warnings that appear on code like this (a pathological example from `ValueNoOverflow.java` test case:

```java
void testShort_plus(@IntRange(from = 0) short x, @IntRange(from = -1) short z) {
        @IntRange(from = 1) short y = (short) (x + 1); // IntRange(from = 0) to IntRange(from = 1)
        @IntRange(from = 0) short w = (short) (z + 1); // GTEN1 to NN
    }
```

I can change the comment here to explain this better.